File: javascript/fibonacci.js
/**
 * Recursive Fibonacci implementation.
 */
function fibonacciRecursive(n) {
    if (n <= 1) {
        return n;
    }
    return fibonacciRecursive(n - 1) + fibonacciRecursive(n - 2);
}

/**
 * Iterative Fibonacci implementation.
 */
function fibonacciIterative(n) {
    if (n <= 1) {
        return n;
    }
    
    let a = 0, b = 1;
    for (let i = 2; i <= n; i++) {
        [a, b] = [b, a + b];
    }
    return b;
}

/**
 * Run Fibonacci benchmarks.
 */
function runBenchmark() {
    const results = [];
    
    // Recursive fibonacci(35)
    const startTimeRecursive = performance.now();
    const resultRecursive = fibonacciRecursive(35);
    const endTimeRecursive = performance.now();
    const executionTimeRecursive = endTimeRecursive - startTimeRecursive;
    
    results.push({
        testName: 'Fibonacci Recursive (n=35)',
        executionTimeMs: executionTimeRecursive,
        result: resultRecursive
    });
    
    // Iterative fibonacci(40)
    const startTimeIterative = performance.now();
    const resultIterative = fibonacciIterative(40);
    const endTimeIterative = performance.now();
    const executionTimeIterative = endTimeIterative - startTimeIterative;
    
    results.push({
        testName: 'Fibonacci Iterative (n=40)',
        executionTimeMs: executionTimeIterative,
        result: resultIterative
    });
    
    return results;
}

if (require.main === module) {
    const results = runBenchmark();
    results.forEach(result => {
        console.log(`Test: ${result.testName}`);
        console.log(`Result: ${result.result}`);
        console.log(`Execution time: ${result.executionTimeMs.toFixed(2)} ms`);
        console.log();
    });
}

module.exports = { runBenchmark };

File: python/fibonacci.py
import time

def fibonacci_recursive(n):
    """
    Recursive Fibonacci implementation.
    """
    if n <= 1:
        return n
    return fibonacci_recursive(n - 1) + fibonacci_recursive(n - 2)

def fibonacci_iterative(n):
    """
    Iterative Fibonacci implementation.
    """
    if n <= 1:
        return n
    
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b

def run_benchmark():
    """
    Run Fibonacci benchmarks.
    """
    results = []
    
    # Recursive fibonacci(35)
    start_time = time.time()
    result_recursive = fibonacci_recursive(35)
    end_time = time.time()
    execution_time_recursive = (end_time - start_time) * 1000
    
    results.append({
        'test_name': 'Fibonacci Recursive (n=35)',
        'execution_time_ms': execution_time_recursive,
        'result': result_recursive
    })
    
    # Iterative fibonacci(40)
    start_time = time.time()
    result_iterative = fibonacci_iterative(40)
    end_time = time.time()
    execution_time_iterative = (end_time - start_time) * 1000
    
    results.append({
        'test_name': 'Fibonacci Iterative (n=40)',
        'execution_time_ms': execution_time_iterative,
        'result': result_iterative
    })
    
    return results

if __name__ == '__main__':
    results = run_benchmark()
    for result in results:
        print(f"Test: {result['test_name']}")
        print(f"Result: {result['result']}")
        print(f"Execution time: {result['execution_time_ms']:.2f} ms")
        print()
